lib/elements/element.py

```diff
@@ -454,6 +454,9 @@ def _get_cache_key(self, previous_stitch):
         cache_key_generator.update(self._get_patterns_cache_key_data())
         cache_key_generator.update(self._get_guides_cache_key_data())
 
+        if hasattr(self, 'element_name') and self.element_name == 'Clone':
```

Comment:
Hmmm, it feels a little weird having EmbroideryElement have to know about something specific to its subclass Clone.  I had something originally in the stitch plan caching PR that would help with this, although I ended up removing it.  Perhaps it's time to bring it back.

I had a method in EmbroideryElement named `get_extra_cache_key_data()`.  In EmbroideryElement it returned nothing, and subclasses could override it to return whatever extra data they liked.  In this case Clone knows that the clone source is relevant, so it would override `get_extra_cache_key_data()` to return the clone source.  What do you think?

Reply:
Sounds good!

Reply:
I decided to just call it `get_cache_key_data()`.  I also noticed that if I changed the color of the clone source, the cache key didn't change, so the clone would use the cache with the old color.  Now it uses the clone sources' cache keys so that we're sure we get everything we care about.